lib/datasource/packagist/index.ts

```diff
@@ -256,15 +264,20 @@ async function packageLookup(
     if (includesPackages?.[name]) {
       return includesPackages[name];
     }
-    if (!providerPackages?.[name]) {
-      return null;
+    let pkgUrl;
+    if (providersLazyUrl) {
```

Comment:
Shouldn't we fist check if package is already available in provider packages and if not heck for lazy provider url?

Reply:
An unlikely combination but probably still the right logic

Reply:
I think it actually is not complete fix, it seems we need some additional stuff added to extract part. It is able to recognize and connect to lazy repositories right now but it doesn't see any updates to packages (as I understood it is extract.ts part)

Reply:
If extract needs to be improved then ideally have it as a separate PR to be merged first, if it can be fixed independently.

Reply:
it seems it works at least on my sample repository it is able to produce PR, https://github.com/maksimru/composer-lazy-repository/pull/1